client/modules/IDE/components/FileNode.jsx

```diff
@@ -190,234 +184,185 @@ class FileNode extends React.Component {
-
-  hideFileOptions = () => {
-    this.setState({ isOptionsOpen: false });
+    setIsOptionsOpen(!isOptionsOpen);
   };
 
-  showEditFileName = () => {
-    this.setState({ isEditingName: true });
-  };
-
-  hideEditFileName = () => {
-    this.setState({ isEditingName: false });
-  };
-
-  showFolderChildren = () => {
-    this.props.showFolderChildren(this.props.id);
-  };
+  const itemClass = classNames({
+    'sidebar__root-item': name === 'root',
+    'sidebar__file-item': name !== 'root',
+    'sidebar__file-item--selected': isSelectedFile,
+    'sidebar__file-item--open': isOptionsOpen,
+    'sidebar__file-item--editing': isEditingName,
+    'sidebar__file-item--closed': isFolderClosed
+  });
 
-  hideFolderChildren = () => {
-    this.props.hideFolderChildren(this.props.id);
-  };
+  const isFile = fileType === 'file';
+  const isFolder = fileType === 'folder';
+  const isRoot = name === 'root';
 
-  renderChild = (childId) => (
-    <li key={childId}>
-      <ConnectedFileNode
-        id={childId}
-        parentId={this.props.id}
-        canEdit={this.props.canEdit}
-        onClickFile={this.props.onClickFile}
-      />
-    </li>
-  );
+  const { extension } = parseFileName(name);
 
-  render() {
-    const itemClass = classNames({
-      'sidebar__root-item': this.props.name === 'root',
-      'sidebar__file-item': this.props.name !== 'root',
-      'sidebar__file-item--selected': this.props.isSelectedFile,
-      'sidebar__file-item--open': this.state.isOptionsOpen,
-      'sidebar__file-item--editing': this.state.isEditingName,
-      'sidebar__file-item--closed': this.props.isFolderClosed
-    });
-
-    const isFile = this.props.fileType === 'file';
-    const isFolder = this.props.fileType === 'folder';
-    const isRoot = this.props.name === 'root';
-
-    const { t } = this.props;
-    const { extension } = parseFileName(this.props.name);
-
-    return (
-      <div className={itemClass}>
-        {!isRoot && (
-          <div
-            className="file-item__content"
-            onContextMenu={this.toggleFileOptions}
-          >
-            <span className="file-item__spacer"></span>
-            {isFile && (
-              <span className="sidebar__file-item-icon">
-                <FileTypeIcon
-                  fileExtension={extension}
+  return (
+    <div className={itemClass}>
+      {!isRoot && (
+        <div className="file-item__content" onContextMenu={toggleFileOptions}>
+          <span className="file-item__spacer"></span>
+          {isFile && (
+            <span className="sidebar__file-item-icon">
+              <FileTypeIcon
+                fileExtension={extension}
+                focusable="false"
+                aria-hidden="true"
+              />
+            </span>
+          )}
+          {isFolder && (
+            <div className="sidebar__file-item--folder">
+              <button
+                className="sidebar__file-item-closed"
+                onClick={showFolderChildren(id)}
```

Comment:
I think the issue with this file is stemming from lines 248 and 260! I think they should've been passing a function instead of an object (i.e `onClick={() => showFolderChildren(id)}` vs. `onClick={showfolderChildren(id)}`.